controllers/comparators/peerAuthentication_comparator.go

```diff
@@ -0,0 +1,17 @@
+package comparators
+
+import (
+	istiosecv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetPeerAuthenticationComparator() ResourceComparator {
+	return func(deployed client.Object, requested client.Object) bool {
+		deployedPeerAuthentication := deployed.(*istiosecv1beta1.PeerAuthentication)
+		requestedPeerAuthentication := requested.(*istiosecv1beta1.PeerAuthentication)
+		return reflect.DeepEqual(deployedPeerAuthentication.Spec.Selector, requestedPeerAuthentication.Spec.Selector) &&
+			reflect.DeepEqual(deployedPeerAuthentication.Spec.Mtls, requestedPeerAuthentication.Spec.Mtls) &&
+			reflect.DeepEqual(deployedPeerAuthentication.Spec.PortLevelMtls, requestedPeerAuthentication.Spec.PortLevelMtls)
```

Comment:
Are these the only fields needed to compare?

Reply:
My thought process behind field wise comparison is to just compare those fields for which we are responsible to manage. If we compare complete object spec then it imposes unnecessary constraints to the user by not allowing him to make any additional modification to it.
Like in the above spec, Our operator is responsible for defining the `Selector`, `Mtls`, `PortLevelMtls` fields. Apart from these field users could make any modification to `PeerAuthentication` like they could add any labels, annotation etc and we simply don't care.